app/uk/gov/hmrc/gform/models/helpers/Javascript.scala

```diff
@@ -33,20 +33,22 @@ object Javascript {
     Future.sequence(fieldIdWithExpr.map(x => toJavascriptFn(x._1, x._2, groupList))).map(_.mkString("\n"))
   }
 
-  def toJavascriptFn(fieldId: FieldId, expr: Expr, groupList: Future[List[List[List[FieldValue]]]]): Future[String] = {
+  def toJavascriptFn(fieldId: FieldId, expr: Expr, groupList: Future[List[List[List[FieldValue]]]])(implicit ex: ExecutionContext): Future[String] = {
 
     expr match {
       case Sum(FormCtx(id)) =>
-        val eventListeners = Group.getGroup(groupList, FieldId(id)).map { x => x.map(y =>
-          s"""document.getElementById("$x").addEventListener("change",sum$id);
+        val eventListeners = Group.getGroup(groupList, FieldId(id)).map { x =>
```

Comment:
I think it would help to understand the code better if the val names were not just x, y, z but names that give the idea of the data you're dealing with, e.g. fieldName, group, type